assets/app/scripts/filters/resources.js

```diff
@@ -845,4 +845,50 @@ angular.module('openshiftConsole')
 
       return portDisplayValue(servicePort.port, servicePort.targetPort, servicePort.protocol);
     };
+  })
+  .filter('routeIngressCondition', function() {
+    return function(ingress, type) {
+      if (!ingress) {
+        return null;
+      }
+      var found = null;
+      angular.forEach(ingress.conditions, function(condition) {
+        if (condition.type === type) {
+          found = condition;
+        }
+      });
+      return found;
+    };
+  })
+  .filter('routeStatus', function() {
+    return function(route) {
+      if (!route.status.ingress) {
+        return "Pending";
+      }
+      angular.forEach(route.status.ingress, function(ingress) {
+        angular.forEach(ingress.conditions, function(condition) {
+          if (condition.type === "Admitted") {
+            return condition.status === "True" ? "Admitted" : "Rejected";
```

Comment:
Yeah... changing this to make use of the routeIngressCondition filter
anyway.

On Wed, Feb 17, 2016 at 2:38 PM, Sam Padgett <notifications@github.com>
wrote:

> In assets/app/scripts/filters/resources.js
> <https://github.com/openshift/origin/pull/7325#discussion_r53218821>:
>
> > +        if (condition.type === type) {
> > +          found = condition;
> > +        }
> > +      });
> > +      return found;
> > +    };
> > +  })
> > +  .filter('routeStatus', function() {
> > +    return function(route) {
> > +      if (!route.status.ingress) {
> > +        return "Pending";
> > +      }
> > +      angular.forEach(route.status.ingress, function(ingress) {
> > +        angular.forEach(ingress.conditions, function(condition) {
> > +          if (condition.type === "Admitted") {
> > +            return condition.status === "True" ? "Admitted" : "Rejected";
>
> Looks like this will just return from the forEach and not the filter?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7325/files#r53218821>.
>